src/trie.rs

```diff
@@ -16,7 +16,7 @@ pub trait Trie<C: NodeCodec, D: DB> {
     fn contains(&self, key: &[u8]) -> TrieResult<bool, C, D>;
 
     /// inserts value into trie and modifies it if it exists
-    fn insert(&mut self, key: &[u8], value: &[u8]) -> TrieResult<(), C, D>;
+    fn insert(&mut self, key: &[u8], value: Vec<u8>) -> TrieResult<(), C, D>;
 
```

Comment:
This modification will result in API incompatibility!

CITA-VM depended on this library and I am not sure if @laizy or more people are using this library.

@u2

Reply:
I am ok with that, but I think the original api is more ergonomic.  optimization and refactoring should be kept internal in my opinion.

Reply:
After check other implementations, 

- https://github.com/paritytech/trie/blob/master/memory-db/src/lib.rs#L316
- https://github.com/paritytech/parity-common/blob/626da48dc2768773bd53a3d019d8fdce471541ab/kvdb/src/lib.rs#L87
- https://docs.rs/rocksdb/0.12.0/src/rocksdb/db.rs.html#1529-1547

I will change this back soom, thanks.